core/src/main/scala/org/apache/spark/rdd/CoalescedRDD.scala

```diff
@@ -58,7 +58,7 @@ private[spark] case class CoalescedRDDPartition(
       val parentPreferredLocations = rdd.context.getPreferredLocs(rdd, p.index).map(_.host)
       preferredLocation.exists(parentPreferredLocations.contains)
     }
-    if (parents.size == 0) 0.0 else (loc.toDouble / parents.size.toDouble)
```

Comment:
Do you mean this is not necessary? It should improve the readability and performance a little bit.